lib/logthrdestdrv.c

```diff
@@ -639,7 +639,8 @@ _worker_thread(gpointer arg)
   _schedule_restart(self);
   iv_main();
 
-  log_threaded_dest_worker_flush(self);
+  worker_insert_result_t result = log_threaded_dest_worker_flush(self);
+  _process_result(self, result);
```

Comment:
Using the `_process_result` has a few drawback, as it could trigger `_disconnect` or `_rewind_batch`. Although they are fine, as both of them should handle consecutive calls.

But it has also a tricky side effect about logs. The logs in the `_process_result` was not composed to use in this case.
For example, if the destination is not connected while the thread is in shutdown, this is going to print on info level the following: `Server disconnected while preparing messages for sending, trying again`. That could be confusing to see.

I did not requested change, as I think the issue it solves outweights the above small inconveniences.